### PR TITLE
Add e2e framework for launching e2e tests on a cluster

### PR DESCRIPTION
**Pull request by kargakis**
This PR imports the ginkgo framework and makes use of it with a preliminary node label test. No cluster creation is handled by the test framework yet, my intention is to reuse createorupdate code for that.

Ref https://jira.coreos.com/browse/AZURE-114

@openshift/sig-azure 
/cc cjryan

### CODE REVIEW

File: test/e2e/osa_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	cs *api.OpenShiftManagedCluster
+}
+
+var c testClient
+
+var _ = BeforeSuite(func() {
+	cwd, _ := os.Getwd()
+	// The current working dir of these tests is down a few levels from the root of the project.
+	// We should traverse up that path so we can find the _data dir
+	configPath := filepath.Join(cwd, "../../_data/containerservice.yaml")
```
Comment (jim-minter) [inline test/e2e/osa_test.go]: (surely no need - just `ioutil.ReadFile("../../blah")`)

**Comment by kargakis** [comment]
rebased, reapplying lgtm

**Pull request merged**